### ISSUE TITLE
X-Forwarded-Proto header is ignored in sabnzbd

### ISSUE THREAD

**Issue by daften** [sabnzbd/sabnzbd#1610]
I've updated sabnzbd  which I use through the linuxserver docker image, and set X-Forwarded-Proto to https in my reverse proxy config. This is not picked up however. My reverse proxy config looks okay and I can't see what else I could do:

```
server {
    listen 9000 ssl http2;
    listen [::]:9000 ssl http2;

    server_name <server_name;

    ssl_certificate ......./fullchain.pem;

    ssl_certificate_key ....../privkey.pem;

    location / {

        proxy_connect_timeout 60;

        proxy_read_timeout 60;

        proxy_send_timeout 60;

        proxy_intercept_errors off;

        proxy_http_version 1.1;

        proxy_set_header        X-Forwarded-Proto            https;

        proxy_set_header        Host            $http_host;

        proxy_set_header        X-Real-IP            $remote_addr;

        proxy_set_header        X-Forwarded-For            $proxy_add_x_forwarded_for;

        proxy_pass http://localhost:8080;

    }

}
```

Am I missing something, or is this indeed a bug?

**Comment by Safihre** [comment]
We don't implement it, not even sure what it is supposed to do.
What is going wrong?

**Comment by daften** [comment]
I'm sorry, let me explain.

My reverse proxy is used for SSL offloading. But when any redirect needs to happen from sabnzbd, the protocol is not taken into account. The config line `proxy_set_header        Host            $http_host;` makes sure the host is still correct, but when I go to https://<hostname>:9000 I get redirected to http://<hostname>:9000/login instead of https://<hostname>:9000/login. This used to work correctly in sabnzbd 2.x before upgrading.

**Comment by Safihre** [comment]
Aaah yes. That's fixed in 3.1.0. You can get it if you switch to the linuxserver testing Docker.

**Comment by daften** [comment]
Excellent, thank you so much for the feedback @Safihre !
